### DISCUSSION TITLE
Longhorn S3 backup retain limit

### DISCUSSION THREAD

**Discussion by cedenilla** [longhorn/longhorn#5710] [Q&A] [answered]
Hello,

We installed Longhorn using Helm and configured S3 backups. However, we have encountered an issue where we are unable to retain more than 50 days of backups due to some limitations.

I have a few questions:

- Is it possible to modify this retention limit?
- Can we add a dynamic variable in Helm that would allow us to adjust the limit as needed?
- Could you please point me to the exact location in Longhorn where the 50-day limit is enforced?
- Is there a specific reason why the limit was set to 50 days?

Thank you for your help,
Daniel

**Comment by cedenilla** [comment]
I created this [PR](https://github.com/longhorn/longhorn-manager/pull/1827)

**Reply by mantissahz** [reply, marked as answer]
1. PR had merged. @cedenilla Thank you.
2. Here is the feature for the dynamic setting. https://github.com/longhorn/longhorn/issues/5713
3. I think retain number is 50 for the maximum number of backups of a recurring job and it is not about days.

Maybe we can close this discussion, WDYT @cedenilla?

**Reply by cedenilla** [reply]
@mantissahz you are right, it's not about days but about number of backups (my misunderstanding). We can close the discussion, with my PR merged we can add more retention if needed without causing performance issues. Thanks a lot!
